hardware_interface/src/resource_manager.cpp

```diff
@@ -63,8 +63,10 @@ class ResourceStorage
     // e.g.: <package_vendor>/<system_type>
     auto interface = std::unique_ptr<HardwareInterfaceT>(
       loader.createUnmanagedInstance(hardware_info.hardware_class_type));
-    HardwareT actuator(std::move(interface));
-    container.emplace_back(std::move(actuator));
+    HardwareT hardware(std::move(interface));
+    container.emplace_back(std::move(hardware));
```

Comment:
```suggestion
    container.emplace_back(HardwareT(std::move(interface)));
```
If we don't need to capture this instance of `hardware` locally, I'd collapse this into a single line and remove the one above (just eyeballed the code for this, please check if this exactly works)

Reply:
Unfortunately, it doesn't work...